Src/Plugins/AzureDevOpsConfiguration/AzureDevOpsConfiguration.SharedStrings.txt

```diff
@@ -0,0 +1,28 @@
+﻿#
+# This file contains key values pairs which are used to expand variable references in
+# the JSON-based regex pattern definition files. This is useful both to share common
+# regex patterns across multiple rules but also to avoid the complexities of escaping
+# JSON string literals. The strings below are directly testable in web sites such as 
+# reg101.com.
+#
+# Shared strings can be assembled even within this file as long as any shared pattern
+# in a strings only contains variables which are defined earlier in the file. So, 
+# a pattern like this is ok (but these lines in reverse order are not).
+#
+# $MyExample=[a-z]
+# $MyExpandedExample=($MyExample|[0-9])
+#
+# Blank lines are ignored when parsing, as are lines that begin with a '#' character.
+# Leading spaces are trimmed at parse time, allowing for indentation.
+
+# Common patterns for file allow and deny lists.
+#
+$BuildDefinitionFiles=(?i)(^https?:\/\/dev.azure.com).*(\/_apis\/build\/definitions\/)
+$ServiceConnectionFiles=(?i)(^https?:\/\/dev.azure.com).*(\/_apis\/contribution\/HierarchyQuery)
+
+$AZC101.BuildDefinitionEntityType=(?i)(?P<type>"EtlEntity"\s*:\s*"builddefinitions")
+$AZC102.ServiceConnectionEntityType=(?i)(?P<type>"EtlEntity"\s*:\s*"serviceendpoints")
+
+$AZC101/240.PullRequestTriggertype=(?i)(?P<triggertype>"triggerType"\s*:\s*"pullRequest")
```

Comment:
>PullRequestTriggertype [](http://example.com/codeflow?start=12&length=22)

So, there's no need yet for a dynamic validation component?

Reply:
Currently implemented rules has all required data in CloudMine do not need API calls to verify.

If we decide to go the route calls API to get certain data which not in CloudMine, will work on it. One obstacle is the authentication required to call API, will look further into that.